From 4cb43c165ad964258c32d504fe939a10ef5e1ce3 Mon Sep 17 00:00:00 2001
From: Mark Kettenis <kettenis@openbsd.org>
Date: Tue, 2 May 2023 20:41:01 +0200
Subject: [PATCH 1/1] env: apple: Enable ENV_IS_IN_FAT

Now that we have a way to find the appropriate EFI system
partition on the internal NVMe storage, store the U-Boot
environment there.

Signed-off-by: Mark Kettenis <kettenis@openbsd.org>
---
 arch/arm/mach-apple/board.c | 10 ++++++++++
 env/Kconfig                 |  1 +
 2 files changed, 11 insertions(+)

diff --git a/arch/arm/mach-apple/board.c b/arch/arm/mach-apple/board.c
index 169d49e8f33..4b36bc7d125 100644
--- a/arch/arm/mach-apple/board.c
+++ b/arch/arm/mach-apple/board.c
@@ -907,6 +907,16 @@ static char *asahi_esp_devpart(void)
 	return devpart;
 }
 
+const char *env_fat_get_intf(void)
+{
+	return blk_get_uclass_name(UCLASS_NVME);
+}
+
+char *env_fat_get_dev_part(void)
+{
+	return asahi_esp_devpart();
+}
+
 #define KERNEL_COMP_SIZE	SZ_128M
 
 #define lmb_alloc(size, addr) lmb_alloc_mem(LMB_MEM_ALLOC_ANY, SZ_2M, addr, size, LMB_NONE)
diff --git a/env/Kconfig b/env/Kconfig
index 7abd82ab6f3..877cd3540ae 100644
--- a/env/Kconfig
+++ b/env/Kconfig
@@ -109,6 +109,7 @@ config ENV_IS_IN_EEPROM
 config ENV_IS_IN_FAT
 	bool "Environment is in a FAT filesystem"
 	depends on !CHAIN_OF_TRUST
+	default y if ARCH_APPLE
 	default y if ARCH_BCM283X
 	default y if ARCH_SUNXI && MMC
 	default y if MMC_OMAP_HS && TI_COMMON_CMD_OPTIONS
